pkg/notifier/email.go

```diff
@@ -0,0 +1,53 @@
+package notifier
+
+import (
+	"fmt"
+	"github.com/go-mail/mail"
+	"github.com/krzysztof-gzocha/curnot/pkg/config"
+)
+
+const NameEmailNotifier = "email"
+
+type Email struct {
+	dialer               *mail.Dialer
+	receiverParameters   *config.EmailReceiverParameters
+	connectionParameters *config.EmailConnectionParameters
+}
+
+func NewEmail(
+	dialer *mail.Dialer,
+	receiverParameters *config.EmailReceiverParameters,
+	connectionParameters *config.EmailConnectionParameters) *Email {
+	return &Email{
+		dialer:               dialer,
+		receiverParameters:   receiverParameters,
+		connectionParameters: connectionParameters,
+	}
+}
+
+func (e *Email) Notify(msg string) error {
+	fmt.Println("Sending email")
+
+	if e.dialer == nil {
+		panic(fmt.Sprintf("Dialer hasn't been set"))
+	}
+
+	message := e.getMessage(msg)
+
+	if err := e.dialer.DialAndSend(message); err != nil {
+		panic(err)
```

Comment:
I think it would be much easier to define some `EmailError` or even `NotifierError` constant, which could have been returned in here instead of error, so the error is treated as [regular value](https://blog.golang.org/errors-are-values). In such scenario we will open the possibility to somehow react to this error instead of just panic. Of course we can use `recover` function, but I think better approach would be to treat the error as regular value as mentioned in a link above.